### PR TITLE
Parse python sources for stdlib list all the way back to 2.3

### PR DESCRIPTION
**Pull request by thatch**
For comparisons against other choices, see

* https://gist.github.com/thatch/1c7139c9946a4e7d6cd65735f000c48e (vs stdlib-list)
* https://gist.github.com/thatch/ebd7e1e94b13104753ba0f5e6455cc12 (vs isort 2.7)
* https://gist.github.com/thatch/f624dac8b4c63cebbc89836dd5d14574 (vs isort 3.9; yes sre removal is real)

### CODE REVIEW

File: stdlibs/__init__.py [hunk 1]
```diff
@@ -6,315 +6,52 @@
...
+        version = "%d%d" % sys.version_info[:2]
+        modname = f"stdlibs.py{version}"
+    elif version == ALL:
+        modname = "stdlibs.py"
+    else:
+        version = "".join(version.split(".")[:2])
+        modname = f"stdlibs.py{version}"
+
+    __import__(modname)
+    return sys.modules[modname].top_level_names  # type: ignore
```
Comment (jreese) [inline stdlibs/__init__.py:49]: Why not `return importlib.import_module(modname)`?

**Comment by jreese** [comment]
Also, it looks like the generated py* modules aren't included?

File: stdlibs/__init__.py (hunk 1 above)
Comment (thatch) [reply stdlibs/__init__.py]: When did we get that newfangled thing.  I don't want to use methods out of stdlib until they've been in there a full decade.  That one's only been there 9 years.

**Comment by thatch** [comment]
I still think we should resolve the offline discussion about these being "modules" -- "top-level names" is technically correct, but both wordy and I don't think most people would understand unless they're involved in packaging.

Here's the current diff vs 3.10 (`stdlibs.old` is prior to this PR; I don't have a copy of 3.10 handy on this machine).

```pycon
>>> stdlibs.old.module_names  - stdlibs.py310.module_names
frozenset()
>>> stdlibs.py310.module_names - stdlibs.old.module_names
frozenset({'_ctypes_test', 'xxsubtype', '_testmultiphase_nonascii_latin', '_testbuffer', '_scproxy', 'xx', '_xxsubinterpreters', '__phello__', '_overlapped', '_testimportmultiple', '_testmultiphase', 'xxlimited_35', '_testconsole', '_xxtestfuzz', '_testcapi', '_testinternalcapi', 'xxlimited'})
```

The names `_ctypes_test`, `_testcapi` and `__phello__` are importable on my Arch install of 3.8.  If they're not considered "stdlib" I'm very confused.

File: README.md [hunk 2]
```diff
@@ -2,6 +2,46 @@
...
+>>> old_ver = stdlibs.stdlib_module_names("3.6")
+>>> "dataclasses" in old_ver
+False
+>>> less_old_ver = stdlibs.stdlib_module_names("3.7")
+>>> "dataclasses" in old_ver
+True
+>>> [v for v in stdlibs.KNOWN_VERSIONS
+...  if "dataclasses" in stdlibs.stdlib_module_names(v)]
+['3.7', '3.8', '3.9', '3.10']
+```
```
Comment (jreese) [inline README.md:43]: This changed with the move to modules. Maybe suggest importing/using the py3x submodules directly unless alongside version string function?

File: stdlibs/__init__.py [hunk 3]
```diff
@@ -6,315 +6,50 @@
...
+    "3.5",
+    "3.6",
+    "3.7",
+    "3.8",
+    "3.9",
+    "3.10",
+]
+
+
+def stdlib_module_names(version: Optional[str] = None):
```
Comment (jreese) [inline stdlibs/__init__.py:39]: Add return type, `typing.FrozenSet[str]`?

File: stdlibs/fetch.py [hunk 4]
```diff
@@ -0,0 +1,251 @@
+import re
+import subprocess
```
Comment (jreese) [inline stdlibs/fetch.py:2]: Add a copyright header?

**Review by jreese — APPROVED** [review]
Docs need updating, and fetch needs a copyright header, but otherwise I'm 👍 on this. If you want to squash or whatever, go for it and let me know when you're ready for a merge.

File: stdlibs/__init__.py (hunk 3 above)
Comment (thatch) [reply stdlibs/__init__.py:39]: Until you enforce this in config, it's going to keep happening... https://github.com/python/mypy/issues/5205

**Pull request merged**